.github/workflows/test.yml

```diff
@@ -0,0 +1,53 @@
+on:
+  push:
+    branches: [master]
+  pull_request:
+name: cargo test
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    name: ${{ matrix.toolchain }}
+    strategy:
+      matrix:
+        toolchain: [stable, beta, nightly]
+    steps:
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.toolchain }}
+      - uses: actions/checkout@v2
+      - run: |
+          chmod 600 .test-key
+          ssh -i .test-key -v -p 2222 -l test-user 127.0.0.1 -o StrictHostKeyChecking=accept-new whoami
+        name: Test ssh connectivity
+      - run: |
+          eval $(ssh-agent)
+          echo "SSH_AUTH_SOCK=$SSH_AUTH_SOCK" >> $GITHUB_ENV
+          echo "SSH_AGENT_PID=$SSH_AGENT_PID" >> $GITHUB_ENV
+          cat .test-key | ssh-add -
+        name: Set up ssh-agent
+      - name: cargo test
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --all-features
+        env:
+          # makes all the ignored tests not ignored
+          RUSTFLAGS: --cfg=ci
+      - run: docker logs $(docker ps | grep openssh-server | awk '{print $1}')
+        name: ssh container log
+        if: ${{ failure() }}
+      - run: docker exec $(docker ps | grep openssh-server | awk '{print $1}') ls -R /config/logs/
+        if: ${{ failure() }}
+      - run: docker exec $(docker ps | grep openssh-server | awk '{print $1}') cat /config/logs/openssh/current
+        name: ssh server log
+        if: ${{ failure() }}
+    services:
+      openssh:
+        image: linuxserver/openssh-server:amd64-latest
+        ports:
+          - 2222:2222
+        env:
+          USER_NAME: test-user
```

Comment:
`linuxserver/openssh-server` does not enable ssh port forwarding by default, it requires an additional environment variable to be set:

```
DOCKER_MODS='linuxserver/mods:openssh-server-ssh-tunnel'
```